packages/rollup-core/src/app/data/consumers/canonical-chain-batch-submitter.ts

```diff
@@ -201,4 +142,53 @@ export class CanonicalChainBatchSubmitter extends ScheduledTask {
 
     return txs
   }
+
+  /**
+   * Waits for a confirm to indicate that the transaction did not fail.
+   *
+   * @param txHash The tx hash to wait for.
+   * @param txBatch The rollup batch in question.
+   * @returns true if the tx was successful and false otherwise.
+   */
+  private async waitForProofThatTransactionSucceeded(
+    txHash: string,
+    txBatch: TransactionBatchSubmission
+  ): Promise<boolean> {
+    try {
+      const receipt: TransactionReceipt = await this.canonicalTransactionChain.provider.waitForTransaction(
+        txHash,
+        1
```

Comment:
One concern to watch out for here is that (at least from a generic EVM perspective) this does not necessarily guarantee that the transaction will not have failed if it has a status of `FINALIZED`, because a 2-block reorg could technically make it fail only after this `await`.

Not sure how hard we want to drill into this sort of failure case ATM, but thought I'd surface.  One potential thing to do here would be to store the `receipt` blocknumber and blockhash, and check for equality once `FINALIZED`, but not really sure if best solution.

Reply:
My concern in the submitter is just to make sure it has been submitted and, if run, it will succeed. A confirm should do this. Whether or not it actually gets mined is more of a concern of the finalizer

That said, yes, in the reorg / censorship case, our clock skew will make the tx fail after the fact. If that's the case, manual intervention will probably be required in the near term, as we'll have to revert geth, delete all unconfirmed outputs, replay transactions, resubmit.

Reply:
Just noting that in the case listed at the end there, we should get repeated loud alerts from [this line](https://github.com/ethereum-optimism/optimism-monorepo/pull/221/files#diff-efe314718d26ffaba472c131bafac2ceR95) (and the same line in the other file).

We'll definitely have to come up with a playbook for these sort of predictable but not preventable cases.